pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchgit
+, avahi
+, boost
+, cmake
+, cudaPackages
+, eigen
+, ffmpeg
+, freetype
+, git
+, glm
+, glslang
+, harfbuzz
+, libdrm
+, libva
+, libpulseaudio
+, monado
+, nlohmann_json
+, onnxruntime
+, openxr-loader
+, pkg-config
+, python3
+, shaderc
+, spdlog
+, systemd
+, udev
+, vulkan-headers
+, vulkan-loader
+, vulkan-tools
+, x264
+, xorg
+}:
+let
+  vendorMonado = monado.overrideAttrs rec {
+    version = "57e937383967c7e7b38b5de71297c8f537a2489d";
+
+    src = fetchgit {
+      url = "https://gitlab.freedesktop.org/monado/monado.git";
+      rev = "${version}";
+      hash = "sha256-O/Td2WccTr4Fa8U64/lVfnidSIH5t3gWuFXCsEVf7bk=";
+    };
+
+    postInstall = ''
+      mv src/xrt/compositor/libcomp_main.a $out/lib/libcomp_main.a
+    '';
+  };
```

Comment:
Is it truly necessary to pin the version? Would this not work if we kept the old version but added the `postInstall`?

Reply:
The dev of WiVRn themself said to use the exact version they have in the CMakeList because it works. I had originally used a different version but had build problems that were solved by pinning.

Reply:
The `let ... in` approach breaks dependency injection, makes it impossible ("hard") to override wivrn derivation in an overlay

> I had originally used a different version but had build problems that were solved by pinning

Fair enough